### DISCUSSION TITLE
Tutorial to help people contributing

### DISCUSSION THREAD

**Discussion by DominiqueMakowski** [easystats/easystats#94] [Show and tell]
Following our recent discussions on how to be more contributor friendly etc., I added this draft of a "crash course" to R development. I realized that, while more and more know how to *use* R (to get stuff done), few actually know how to develop or program in it, which is particularly sad given that IMO the entry barrier is quite low: there are only a few concepts to be aware of (e.g., how methods work) to start programming and contributing to R projects.

https://easystats.github.io/easystats/articles/programming.html

What else do you think would be relevant?

**Comment by mattansb** [comment]
- Documenting a function with roxygen.
- github: What are PRs (and what is CI).
- If you're aiming for `easystats` style, I would talk about exported vs. unexported (here: internal support functions), and how everything is integrated in one way or another to `insight`.
